common/persistence/dataInterfaces.go

```diff
@@ -821,11 +832,11 @@ type (
 		NextPageToken []byte
 	}
 
-	// SerializedHistoryEventBatch represents a serialized batch of history events
-	SerializedHistoryEventBatch struct {
-		EncodingType common.EncodingType
-		Version      int
-		Data         []byte
+	// DataBlob represents a blob for any binary data.
+	// The raw data is in Data field, and metadata(encoding/version/compression/etc) is in headers field
+	DataBlob struct {
+		Headers map[string]string
```

Comment:
These are always going to be Cadence specific headers so I think we should just stick to strongly typed interface here.